src/main/java/games/strategy/engine/random/PropertiesDiceRoller.java

```diff
@@ -247,3 +256,28 @@ public String getHelpText() {
     return getInfoText();
   }
 }
+
+
+class AdvancedRedirectStrategy extends LaxRedirectStrategy {
```

Comment:
In general, such classes should be nested private members of the sole type that uses them.  They don't need to be used by any other type in the package, so they shouldn't have package-private accessibility.

However, I don't think this should be done for the delegate extension classes because they appear to be serialized.  Nesting them in their enclosing type would change their name and thus break serialization.

So, as you said @simon33-2, the proper way to fix the violation is to move them as-is to a separate file.  However, this is another case where we can discuss if in the particular case of delegate extension classes, we leave them as they are and simply suppress the warning.